src/NuGet.Clients/NuGet.VisualStudio.Common/IDE/VsHierarchyUtility.cs

```diff
@@ -24,12 +24,21 @@ public static class VsHierarchyUtility
 
         public static string GetProjectPath(IVsHierarchy project)
         {
+            Assumes.Present(project);
             ThreadHelper.ThrowIfNotOnUIThread();
             var format = (IPersistFileFormat)project;
             format.GetCurFile(out string projectPath, out uint _);
             return projectPath;
         }
 
+        public static string GetProjectPathForWebsiteProject(IVsHierarchy project)
+        {
+            Assumes.Present(project);
+            ThreadHelper.ThrowIfNotOnUIThread();
+            project.GetCanonicalName((uint)VSConstants.VSITEMID.Root, out string projectPath);
```

Comment:
Team Review: Please put some comments in here so people know why we are special casing web site projects

Reply:
Added.